### DISCUSSION TITLE
How to set a multilingual Search Center URL (from Site Settings > Search Settings)?

### DISCUSSION THREAD

**Discussion by PzKfWg** [microsoft-search/pnp-modern-search#2955] [Ideas]
Hi! I have a PnP modern Search web part enabled search center for my intranet and I want to redirect the Microsoft Search top bar to this page. I can do this easily from the Site Settings > Search Settings and plug my page there. However, my site is multilingual and I have multiple translated versions of my search page. Even if I put my main language page URL, it does not honor the usual language switching path (/en or /fr) and only render the main language, regardless of the current language.

Does anyone know if there is a way to point to language specific search result page? Thanks!
@FranckyC

**Comment by wobba** [comment]
No, the redirect support one URL only unfortunately.

**Comment by PzKfWg** [comment]
Thanks for the quick answer! I suppose there could be a way to have a classic redirect page that would send the query to the correct language... I'll post here if I find a way.

**Comment by PzKfWg** [comment]
Okay, here is what I did that worked: I added the Modern Script Editor web part at the top of my primary language (French) search page. I enabled the classic _spPagecontextInfo to be able to see the desired SharePoint language. Then I simply checked for most English codes and redirect to the English version and the query string as well. I don't lose any performance on the primary language and it is barely perceptible on the secondary languages. Here is the code:
```
<script>
    if (_spPageContextInfo.currentLanguage == 1033 ||
        _spPageContextInfo.currentLanguage == 2057 ||
        _spPageContextInfo.currentLanguage == 3081 ||
        _spPageContextInfo.currentLanguage == 4105 ||
        _spPageContextInfo.currentLanguage == 5129 ||
        _spPageContextInfo.currentLanguage == 6153 ||
        _spPageContextInfo.currentLanguage == 7177 ||
        _spPageContextInfo.currentLanguage == 16393
        ) {
        var query = new URLSearchParams(window.location.search);
        var url = 'https://[yourtenant].sharepoint.com/SitePages/en/modernSearch.aspx?' + query.toString();
        window.location.href = url;
    }
</script>
```
Thank you so much Mikael! I solved the issue using another of your web parts!

**Reply by wobba** [reply]
I would have liked an app customizer better, or adding the logic into the web part. Use of script editor if others can edit the page is a security risk. I'll tag this as an enhancement ask :)
